src/Orleans/Core/Exceptions.cs

```diff
@@ -29,6 +29,14 @@ protected OrleansException(SerializationInfo info, StreamingContext context)
         }
     }
 
+    public static class OrleansExceptionExtensions
+    {
+        public static bool IsClientNotAvailableException(this OrleansException orleansException)
```

Comment:
Yeah, this sucks.  I had planned to discuss this after this PR, but we can do this now.  :)
The problem is that when the catalog encounters this condition, it rejects the message and rejected messages always come back as OrleansException.  The original exception becomes the message of the Orleans exception.  I was not sure what depended on this behavior so I chose not to change it.  This extension class allowed me to avoid addressing what was potentially a breaking change (how we handle rejection responses) in this PR.  :/

I think the behavior we want would be:
If original exception is of type OrleansException (which all should be), we pass it through.  If it is not, we create an OrleansException with the message and inner exception coming from the oringial exception.

Questions:
Should we address the rejection exception change in this PR?
If so, is the above behavior desirable, or is there better alternative?